auth/src/main/java/com/firebase/ui/auth/compose/ui/screens/FirebaseAuthScreen.kt

```diff
@@ -0,0 +1,603 @@
+                Log.e("FirebaseAuthScreen", "Failed to complete email link sign-in", e)
+            }
+
+            if (navController.currentBackStackEntry?.destination?.route != AuthRoute.Email.route) {
+                navController.navigate(AuthRoute.Email.route)
+            }
+        }
+    }
+
+    // Synchronise auth state changes with navigation stack.
+    LaunchedEffect(authState) {
+        val state = authState
+        val currentRoute = navController.currentBackStackEntry?.destination?.route
+        when (state) {
+            is AuthState.Success -> {
+                pendingResolver.value = null
+                pendingLinkingCredential.value = null
+
+                state.result?.let { result ->
+                    if (state.user.uid != lastSuccessfulUserId.value) {
+                        onSignInSuccess(result)
+                        lastSuccessfulUserId.value = state.user.uid
+                    }
+                }
+
+                if (currentRoute != AuthRoute.Success.route) {
+                    navController.navigate(AuthRoute.Success.route) {
+                        popUpTo(AuthRoute.MethodPicker.route) { inclusive = true }
+                        launchSingleTop = true
+                    }
+                }
+            }
+
+            is AuthState.RequiresEmailVerification,
+            is AuthState.RequiresProfileCompletion -> {
+                pendingResolver.value = null
+                pendingLinkingCredential.value = null
+                if (currentRoute != AuthRoute.Success.route) {
+                    navController.navigate(AuthRoute.Success.route) {
+                        popUpTo(AuthRoute.MethodPicker.route) { inclusive = true }
+                        launchSingleTop = true
+                    }
+                }
+            }
+
+            is AuthState.RequiresMfa -> {
+                pendingResolver.value = state.resolver
+                if (currentRoute != AuthRoute.MfaChallenge.route) {
+                    navController.navigate(AuthRoute.MfaChallenge.route) {
+                        launchSingleTop = true
+                    }
+                }
+            }
+
+            is AuthState.Cancelled -> {
+                pendingResolver.value = null
+                pendingLinkingCredential.value = null
+                lastSuccessfulUserId.value = null
+                if (currentRoute != AuthRoute.MethodPicker.route) {
+                    navController.navigate(AuthRoute.MethodPicker.route) {
+                        popUpTo(AuthRoute.MethodPicker.route) { inclusive = true }
+                        launchSingleTop = true
+                    }
+                }
+                onSignInCancelled()
+            }
+
+            is AuthState.Idle -> {
+                pendingResolver.value = null
+                pendingLinkingCredential.value = null
+                lastSuccessfulUserId.value = null
+                if (currentRoute != AuthRoute.MethodPicker.route) {
```

Comment:
i caught a bug while running e2e tests, if auth state is Idle this throws `Cannot navigate to auth_method_picker. Navigation graph has not been set for
 NavController`.

moving this to after NavHost has been initialized fixed it.

> NavHost(navController = navController, startDestination = AuthRoute.MethodPicker.route)
> 
> // Handle email link sign-in (deep links)
> LaunchedEffect(emailLink) {}
> 
> // Synchronise auth state changes with navigation stack.
> LaunchedEffect(authState) {}
>